### DISCUSSION TITLE
Self-Hosted Supabase: A Missed Opportunity for True Open Source

### DISCUSSION THREAD

**Discussion by picselladmin** [supabase/supabase#17876] [General]
**Introduction**

I was recently charmed by supabase and particularly their auth providers configuration through the studio. Before relying on it, I tried to deploy it to a production environment in a self hosted manner only to discover that crucial features are intentionally restricted in self-hosted setups. This raises serious questions about Supabase's commitment to open-source principles.

**The Issue: Feature Limitations**

The use of the IS_PLATFORM variable seems to deliberately limit essential features in self-hosted instances. This is not an isolated issue; multiple users have raised similar concerns, asking for better documentation to clarify this limitation and were promised _support coming soon_ since 2021. Here are some relevant discussions and issues for reference:

https://github.com/orgs/supabase/discussions/13114
https://github.com/orgs/supabase/discussions/7238
https://github.com/orgs/supabase/discussions/4444
https://github.com/orgs/supabase/discussions/4907
https://github.com/supabase/supabase/issues/12835

**Self-Hosting and Feature Parity**

In my opinion, a self-hosted solution should mirror the features of its managed counterpart, especially when it comes to something as vital as authentication. The current architecture not only fails to meet this expectation but also introduces unnecessary complexity. For instance, authentication providers can be configured via environment variables, but the setup is cumbersome, poorly documented, and insecure by default. If anything, maintaining 2 versions increases the maintenance complexity and, in one of the issues above, generated a bug (still open)

**Conclusion: A Call for Alignment with Open Source Principles**

The current state of Supabase's self-hosted version is a far cry from the ethos of open-source software. It harkens back to the days of **shareware**, where software was distributed for free but with limited functionality, nudging users towards the paid version. I strongly urge the Supabase team to reconsider this approach and align their self-hosted offering more closely with true open-source principles. If I'm missing something, I'm open to being corrected.

Feel free to use this as a basis for your critique or feedback.

**Comment by bicepjai** [comment]
It’s honestly disappointing that this topic hasn’t sparked more engagement over the years.
While I understand that in a capitalist market, it’s natural for there to be feature distinctions between free and paid versions, the way Supabase is marketed as "open source" does raise eyebrows given the limitations imposed on self-hosted deployments. At the very least, the differences between self-hosted and managed versions should be transparently and explicitly articulated before making strong open source claims. Otherwise, it risks undermining trust in the community and the real meaning of open source.

**Reply by kmhari** [reply]
made a control plane backend for the IS_PLATFORM=true version here https://github.com/kmhari/supastack
